### PR TITLE
🎨 Adds new progress bar to track `starting software`

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Backend sends out messages when it's starting the user service containers. Once all docker compose commands have finished the progress bar will indicate 100%

```
42["nodeProgress",{"progress_type":"SERVICE_CONTAINERS_STARTING","progress_report":{"actual_value":3.0,"total":5.0,"unit":null,"message":{"description":"starting software","current":3.0,"total":5,"unit":null,"sub":null}},"node_id":"c2496768-a89e-4fb4-a40d-01826b4221dc","project_id":"12f1fa8a-70e2-11ef-a54b-0242ac140004","user_id":1}]
```

Sep-12-2024 17-10-43

### Bonus

- 🐛 removed double pulling of docker images (shaves off a tiny bit of time during boot)
- 🐛 fixed an issue causing progress to be reported wrongly thanks to @sanderegg  

Progress reported **before** fix: this would translate in the progress being stuck at the max value.

broken

Progress reported **after** fix: progress raises linearly, as expected

fixed

## Related issue/s

- closes https://github.com/ITISFoundation/osparc-issues/issues/1710
- closes https://github.com/ITISFoundation/osparc-issues/issues/1714
- closes https://github.com/ITISFoundation/osparc-issues/issues/1715
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/6156

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by giancarloromeo — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]
as discussed then don't pull twice. very nice full stack!

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js [hunk 1]
```diff
@@ -220,6 +235,7 @@ qx.Class.define("osparc.data.model.NodeProgressSequence", {
       this.__pullingOutputsLayout = sequenceLoadingPage.addNewTask(qx.locale.Manager.tr("Retrieving your output data ..."));
       this.__pullingStateLayout = sequenceLoadingPage.addNewTask(qx.locale.Manager.tr("Retrieving your work ..."));
       this.__pullingImagesLayout = sequenceLoadingPage.addNewTask(qx.locale.Manager.tr("Installing software ..."));
+      this.__startingSoftwareLayout = sequenceLoadingPage.addNewTask(qx.locale.Manager.tr("Starting software ..."));
```
Comment (sanderegg) [inline services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:238]: nTasks?

**Review by sanderegg — APPROVED** [review]
please check my comments regarding the frontend part...

**Review by pcrespov — APPROVED** [review]
:+1:

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py [hunk 2]
```diff
@@ -168,15 +168,28 @@ async def task_create_service_containers(
...
+    async with event_propagation_disabled(app), _reset_on_error(
+        shared_store
+    ), ProgressBarData(
+        num_steps=5,
+        progress_report_cb=functools.partial(
+            post_progress_message,
+            app,
+            ProgressType.SERVICE_CONTAINERS_STARTING,
+        ),
+        description=IDStr("starting software"),
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py:180]: This is strange. `IDStr` is not intended for description but rather string identifiers, note that ID means "identifier". Not sure why it was set a description as an identifier. Please check the other constrainted types already defined around IDStr and see which one might best fix.

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 1 above)
Comment (GitHK) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:238]: not sure what you refer to here

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py (hunk 2 above)
Comment (GitHK) [reply services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py:180]: This one fits perfectly, We want to strip whitespaces and limit the size. Maybe a rename can be done in the future for an alias?

**Review by odeimaiz — APPROVED** [review]

File: packages/service-library/tests/aiohttp/test_docker_utils.py [hunk 3]
```diff
@@ -110,7 +110,11 @@ def _assert_progress_report_values(
 
 @pytest.mark.parametrize(
     "image",
-    ["itisfoundation/sleeper:1.0.0", "nginx:latest"],
+    [
+        "itisfoundation/sleeper:1.0.0",
+        "nginx:latest",
+        # "registry.osparc-master.speag.com/simcore/services/dynamic/jupyter-math:3.0.3",
+    ],
 )
```
Comment (sanderegg) [inline packages/service-library/tests/aiohttp/test_docker_utils.py:118]: please revert that one

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py [hunk 4]
```diff
@@ -194,13 +198,15 @@ async def task_create_service_containers(
...
         await _retry_docker_compose_create(shared_store.compose_spec, settings)
+        await progress_bar.update()
 
         progress.update(
             message="ensure containers are started", percent=ProgressPercent(0.95)
         )
         compose_start_result = await _retry_docker_compose_start(
             shared_store.compose_spec, settings
         )
+        await progress_bar.update()
```
Comment (sanderegg) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py:209]: this is not needed, the context manager does the last one anyway

File: services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js (hunk 1 above)
Comment (sanderegg) [reply services/static-webserver/client/source/class/osparc/data/model/NodeProgressSequence.js:238]: line 231

**Review by sanderegg — APPROVED** [review]
thanks and please check the JS file

**Review by matusdrobuliak66 — APPROVED** [review]

**Review by ignapas — APPROVED** [review]

**Pull request merged**